src/ripple/protocol/impl/Feature.cpp

```diff
@@ -20,17 +20,43 @@
 #include <BeastConfig.h>
 #include <ripple/protocol/digest.h>
 #include <ripple/protocol/Feature.h>
+
+#include <boost/container/flat_map.hpp>
+
 #include <cstring>
 
 namespace ripple {
 
+static boost::container::flat_map<uint256, std::size_t> featureToIndex;
+static std::vector<uint256> features;
```

Comment:
I think we have an problem if `feature()` is invoked after `main()`  by different threads. Of course, the intention is that `feature` is only used to initialize our constants, so the order of init seems safe in this case (this one translation unit only is involved). `feature` is also being called in one of the new unit tests here and multithread access is not needed there. I agree we need to be careful about this - I'd like to hear what @seelabs has to say about this.